packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await expect(
+        escrowClient.fund(escrowAddress, invalidAmount)
+      ).rejects.toThrow(ErrorAmountMustBeGreaterThanZero);
+    });
+
+    test('should successfully fund escrow', async () => {
+      const tokenAddress = ethers.constants.AddressZero;
+      const escrowAddress = ethers.constants.AddressZero;
+      const amount = BigNumber.from(10);
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.token.mockReturnValue(tokenAddress);
+
+      await escrowClient.fund(escrowAddress, amount);
+
+      expect(escrowClient.escrowContract.token).toHaveBeenCalledWith();
+      expect(escrowClient.tokenContract.transfer).toHaveBeenCalledWith(
+        escrowAddress,
+        amount
+      );
+    });
+
+    test('should throw an error if the setup escrow fails', async () => {
+      const tokenAddress = ethers.constants.AddressZero;
+      const escrowAddress = ethers.constants.AddressZero;
+      const amount = BigNumber.from(10);
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.token.mockReturnValue(tokenAddress);
+      escrowClient.tokenContract.transfer.mockRejectedValueOnce(new Error());
+
+      await expect(escrowClient.fund(escrowAddress, amount)).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.token).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('storeResults', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+      const url = VALID_URL;
+      const hash = FAKE_HASH;
+
+      await expect(
+        escrowClient.storeResults(invalidAddress, url, hash)
+      ).rejects.toThrow(ErrorInvalidEscrowAddressProvided);
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'